applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -1,10 +1,152 @@
+    * Configuration file for the bootloader.
+    * DTS overlay file.
+
+nRF Desktop board configuration files
+=====================================
+
+The nRF Desktop project comes with configuration files for the following boards:
+
+Gaming mouse (nrf52840_pca20041)
+      * The board is defined in ``nrf/boards/arm/nrf52840_pca20041`` for the project-specific hardware.
+      * To achieve gaming-grade performance:
+
+        * The application is configured to act as a gaming mouse, with both BLE and USB transports enabled.
+        * Bluetooth is configured to use Nordic's proprietary link layer.
+
+      * |preconfigured_build_types|
+
+Desktop mouse (nrf52_pca20044 and nrf52810_pca20045)
+      * Both boards are meant for the project-specific hardware and are defined in ``nrf/boards/arm/nrf52_pca20044`` and ``nrf/boards/arm/nrf52810_pca20045``, respectively.
+      * The application is configured to act as a mouse.
+      * Only the Bluetooth Low Energy transport is enabled.
+        Bluetooth uses Zephyr's software link layer.
+      * There is no configuration with bootloader available.
+
+
+Sample mouse or keyboard (nrf52840_pca10056)
+      * The configuration uses the nRF52840 Development Kit.
+      * The build types allow to build the application both as mouse and as keyboard.
+      * Inputs are simulated based on the hardware button presses.
+      * The configuration with bootloader is available.
+
+Keyboard (nrf52_pca20037)
+      * The board used is defined in ``nrf/boards/arm/nrf52_pca20037`` for the project-specific hardware.
+      * The application is configured to act as a keyboard, with the BLE transport enabled.
+      * Bluetooth is configured to use Nordic's proprietary link layer.
+      * |preconfigured_build_types|
+
+Dongle (nrf52840_pca10059)
+      * This configuration uses Nordic's nRF52840 dongle defined in Zephyr.
+      * Since the board is generic, project-specific changes are applied in the DTS overlay file.
+      * The application is configured to act as both mouse and keyboard.
+      * Bluetooth uses Nordic's proprietary link layer and is configured to act as a central.
+        Input data come from Bluetooth and are retransmitted to the USB.
+      * |preconfigured_build_types|
+
+.. _porting_guide_adding_board:
+
+Adding a new board
+******************
+
+When adding a new board for the first time, focus on a single configuration.
+Moreover, keep the default ``ZDebug`` build type that the application is built with, and do not add any additional build type parameters.
+
+.. note::
+    The following procedure will use the gaming mouse configuration as example.
+
+To use the nRF Desktop project with your custom board:
+
+1. Define the board by copying the nRF Desktop board files that are the closest match for your hardware.
+   For example, for gaming mouse use ``nrf/boards/arm/nrf52840_pca20041``.
+#. Edit the DTS files to make sure they match the hardware configuration.
```

Comment:
There are also the Kconfig files for the board

Reply:
There are and it is worth mentioning here.

Reply:
done